src/lib/pipeline/ExecuteUnit.cc

```diff
@@ -215,6 +223,56 @@ void ExecuteUnit::purgeFlushed() {
   }
 }
 
+void ExecuteUnit::seqIdFlush(uint64_t afterSeqId) {
+  if (pipeline_.size() == 0) {
+    return;
+  }
+
+  // If the newest instruction has been flushed, clear any stalls.
+  if (pipeline_.back().insn->getSequenceId() > afterSeqId) {
+    stallUntil_ = tickCounter_;
+  }
+
+  // Iterate over the pipeline and remove flushed instructions
+  while (!pipeline_.empty()) {
+    if (pipeline_.back().insn->getSequenceId() <= afterSeqId) {
+      break;
+    }
+    pipeline_.pop_back();
+  }
+
+  // If first blocking in-flight instruction is flushed, ensure another
+  // non-flushed stalled instruction takes it place in the pipeline if
+  // available.
+  bool replace = false;
+  if (operationsStalled_.size() > 0 &&
+      operationsStalled_.front()->getSequenceId() > afterSeqId) {
+    replace = true;
+  }
+  auto itStall = operationsStalled_.begin();
+  while (itStall != operationsStalled_.end()) {
+    auto& entry = *itStall;
+    if (entry->getSequenceId() > afterSeqId) {
+      itStall = operationsStalled_.erase(itStall);
+    } else {
+      itStall++;
+    }
+  }
+
+  if (replace && operationsStalled_.size() > 0) {
+    // Add uop to pipeline
+    auto& uop = operationsStalled_.front();
+    pipeline_.push_back({nullptr, tickCounter_ + uop->getLatency() - 1});
+    pipeline_.back().insn = std::move(uop);
+    operationsStalled_.front() = pipeline_.back().insn;
+  }
```

Comment:
When flushed, a now unblock operation should transition to the internal pipeline straight away. However, if we do it outside of the flush and with the tick, then it could be multiple cycles until it enters the pipeline as that logic is only triggered when the front of the pipeline is executed.